service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs

```diff
@@ -71,9 +106,50 @@ public void ItCanBeDisabledInAnyEnvironment(string environment)
         Assert.False(SensitiveDataLogger.Enabled);
     }
 
+    [Theory]
+    [Trait("Category", "UnitTest")]
+    [InlineData("development")]
+    [InlineData("Development")]
+    [InlineData("staging")]
+    [InlineData("Staging")]
+    [InlineData("production")]
+    [InlineData("Production")]
+    [InlineData("any")]
+    public void ItCanBeDisabledInAnyDotNetEnvironment(string environment)
+    {
+        // Arrange
+        Assert.False(SensitiveDataLogger.Enabled);
+        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, environment);
+
+        // Act
+        SensitiveDataLogger.Enabled = false;
+
+        // Assert
+        Assert.False(SensitiveDataLogger.Enabled);
+    }
+
+    [Theory]
+    [Trait("Category", "UnitTest")]
+    [InlineData("development", "staging")]
+    [InlineData("Development", "Staging")]
+    public void AspNetCoreEnvironmentHasThePrecedence(string aspNetCoreEnvironment, string dotNetEnvironment)
```

Comment:
this doesn't seem to test the precedence, does it?

Reply:
I believe this should be a couple of tests

##1

        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, "development");
        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, "production");
        SensitiveDataLogger.Enabled = true;
         => can be enabled



##2

        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, "production");
        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, "development");
        Assert.Throws<InvalidOperationException>(() => SensitiveDataLogger.Enabled = true);
         => throws

Reply:
> this doesn't seem to test the precedence, does it?

It tests that the `ASPNETCORE_ENVIRONMENT`, that is set to **Development**, takes the precendence over `DOTNET_ENVIRONMENT`, that is set to **Staging**. But of course we can make two separate tests.